src/transformers/trainer.py

```diff
@@ -216,9 +216,16 @@ def __init__(
             model is not None or model_init is not None
         ), "You must provide a model to use `Trainer`, either by using the `model` argument or the `model_init` argument."
         self.model_init = model_init
+
         if model is None and model_init is not None:
             model = self.call_model_init()
-        self.model = model.to(args.device) if model is not None else None
+
+        # Model parallel
+        self.model_parallel = args.model_parallel
```

Comment:
`args` are already stored inside the `Trainer` so you shouldn't make a shortcut that will have people wonder where this `model_parallel` came from. `self.args.model_parallel` is short enough and it's clear from the code how to activate this behavior.
All other fields of `TrainingArguments` are treated the same way.

Reply:
Changed instances of `self.model` parallel to `self.args.model_parallel`.